src/components/message/message.test.js

```diff
@@ -0,0 +1,135 @@
+import * as React from "react";
+import { useState } from "react";
+import Button from "../button";
+import Message from "./message.component";
+import CypressMountWithProviders from "../../../cypress/support/component-helper/cypress-mount";
+
+import {
+  messagePreview,
+  messageChildren,
+  messageTitle,
+  messageDismissIcon,
+  closeButton,
+  variantPreview,
+} from "../../../cypress/locators/message/index";
+
+const testData = ["mp150ú¿¡üßä", "!@#$%^*()_+-=~[];:.,?{}&\"'<>"];
+
+const MessageComponent = ({ ...props }) => {
+  const [isOpen, setIsOpen] = useState(true);
+  return (
+    <div>
+      {!isOpen && <Button onClick={() => setIsOpen(true)}>Open Message</Button>}
+      <Message open={isOpen} onDismiss={() => setIsOpen(false)} {...props}>
+        Some custom message
+      </Message>
+    </div>
+  );
+};
+
+context("Tests for Message component", () => {
+  describe("should check Message component properties", () => {
+    it.each([
+      ["info", "rgb(51, 91, 112)"],
+      ["error", "rgb(205, 56, 75)"],
+      ["success", "rgb(0, 138, 33)"],
+      ["warning", "rgb(239, 103, 0)"],
+    ])(
+      "should check %s variant for Message component",
+      (variant, backgroundColor) => {
+        CypressMountWithProviders(<MessageComponent variant={variant} />);
+        variantPreview().should(
+          "have.css",
+          "background-color",
+          backgroundColor
+        );
+      }
+    );
+
+    it.each(testData)(
+      "should check %s as children for Message component",
+      (stringValue) => {
+        CypressMountWithProviders(<Message>{stringValue}</Message>);
+        messageChildren().should("have.text", stringValue);
+      }
+    );
+
+    it("should set className for Message component", () => {
+      CypressMountWithProviders(
+        <MessageComponent className="message-cypress-classname" />
+      );
+      messagePreview().should("have.class", "message-cypress-classname");
+    });
+
+    it("should check custom id for Message component", () => {
```

Comment:
This test should use the string data type const to test with, similar to the children and title tests, as this can be set with any string.